### ISSUE TITLE
Single message notification could use contact avatar

### ISSUE THREAD

**Issue by woj-tek** [k9mail/k-9#2314]
A follow-up to #2239

### Expected behavior
When there is only a single-item notification the icon in the opened notification could use contact avatar (if available) instead of default icon.

### Actual behavior
We see default icon (please see above issue for screenshots)

### Environment
K-9 Mail version: 5.203

Android version: 6.0.1

Account type (IMAP, POP3, WebDAV/Exchange): n/a

**Comment by Valodim** [comment]
Nice idea! This should also be simple to code, @cketti beginner issue?

**Comment by yashsharan** [comment]
I would like to work on this issue. Any suggestions from where should  i start?

**Comment by Valodim** [comment]
I don't think this issue requires any handholding. Feel free to ask concrete questions if you have some :)

**Comment by yashsharan** [comment]
@Valodim The main problem with this issue is that setSmallIcon method in NotificationCompat only accepts int as an argument(Resource id of the image) and not any other method.So if i try to set the contact picture as image it gives an error of wrong argument type
The option to set image other than using int as an argument is supported only after API 23+

**Comment by cketti** [comment]
The contact picture should go into "large icon". See also https://material.io/guidelines/patterns/notifications.html

**Comment by yashsharan** [comment]
@cketti thanks..i will work on it right away

**Comment by SIMRAN1** [comment]
I would like to work on this issue

**Comment by AllanWang** [comment]
I had a look at this, and I'm just wondering why the Address object checks for equality for `personal` and `address` rather than just the `address`. It didn't seem like the Notifications had access to the address, and initializing a new one without `personal` may not lead to the bitmap retrieval from the cache.

**Comment by cketti** [comment]
While we could change the cache to only use the email address as key, the `personal` value is still required when we have to generate a fallback image. So the right solution is probably to store the sender `Address` in `NotificationContent`.

**Comment by AllanWang** [comment]
That sounds reasonable, but when comparing if two addresses are the same, shouldn't we just use the email? I guess I still haven't sifted through enough code to figure out where the personal values are stored and retrieved, so I haven't found how much of the content I need to change before enabling it

**Comment by Gaelan-Bolger** [comment]
Interested in a PR for this? [Show contact avatar in single message notification #2314](https://github.com/Gaelan-Bolger/k-9/commit/84207e067c86d92a26abd6de53bd01b8a5ed8a43)

**Comment by AllanWang** [comment]
@Gaelan-Bolger  it looks nice to me at least, maybe make a pull request so others can see it? One thing I might suggest is having a circular bitmap for higher android versions to match the design. I also think that sender is the same as the address in `Address`, so having two arguments may be redundant?

**Comment by Gaelan-Bolger** [comment]
Ok, I'll submit a PR for this commit. I left sender as part of NotificationContent because it seems sender and displaySender can sometimes be different? Maybe circle avatar for higher APIs can be a separate beginners issue?

**Comment by rhari991** [comment]
There is a separate issue for it already - #2252
